lib/imsg-buffer.c

```diff
@@ -183,6 +183,8 @@ void msgbuf_drain(struct msgbuf *msgbuf, size_t n)
 		next = TAILQ_NEXT(buf, entry);
 		if (buf->rpos + n >= buf->wpos) {
 			n -= buf->wpos - buf->rpos;
+
+			TAILQ_REMOVE(&msgbuf->bufs, buf, entry);
```

Comment:
I think we need to double check if this change makes sense. Since the imsg framework is used quite extensively in several OpenBSD daemons, I think it's unlikely a bug of this kind would remain unnoticed for so much time. Unlikely but not impossible. I can't analyze this closely now (packing for vacation) but I'd appreciate if you or someone else could take a closer look at this change.

Reply:
Ah, I see you just moved the `TAILQ_REMOVE` from `ibuf_dequeue`... please disregard what I just wrote.